src/lab10.py

```diff
@@ -56,8 +58,9 @@ def request(url, headers={}, payload=None):
     method = "POST" if payload else "GET"
     body = "{} {} HTTP/1.0\r\n".format(method, path)
     body += "Host: {}\r\n".format(host)
-    for header, value in headers.items():
```

Comment:
Why did you delete this part?

Reply:
So, in earlier versions of this chapter we took a map of extra headers as an input to `request`. It was only used for cookies. Now I handle cookies directly in `request` and don't need that argument.

Passing in a headers dict was done incorrectly in edge cases (you could send the Host header twice, for example) and using it for cookies was also hard / incorrect (you need to handle `Cookie`/`Set-Cookie` at every request, which we weren't doing). It also didn't work well with a multi-tab design. Plus it shadowed an exercise in Ch1. I think handling cookies entirely in `request` is better design, cleaner code, and that there's now no reason to support a `headers` argument.